astropy/io/ascii/mrt.py

```diff
@@ -461,46 +471,54 @@ def write(self, lines):
-                        self.cols.append(lon_col)
-                        self.cols.append(lat_col)
-
-                    # Ecliptic coordinates, can be any of various available.
-                    elif 'ecliptic' in col.name:
-                        lon_col = Column(col.lon, name='ELON',
-                                         description='Ecliptic Longitude (' + col.name + ')',
-                                         unit=col.representation_component_units['lon'],
-                                         format='.12f')
-                        lat_col = Column(col.lat, name='ELAT',
-                                         description='Ecliptic Latitude (' + col.name + ')',
-                                         unit=col.representation_component_units['lat'],
-                                         format='.12f')
-                        self.cols.append(lon_col)
-                        self.cols.append(lat_col)
-
-                    # Convert all other ``SkyCoord`` columns that are not in the above three
-                    # representations to string valued columns.
-                    else:
-                        self.cols.append(Column(col.to_string()))
-
-                to_pop.append(i)   # Delete original ``SkyCoord`` column.
+                    frminfo = ''
+                    for frame, latlon in coord_systems.items():
+                        if frame in col.name and len(set(latlon[:2]) - set(self.colnames)) == 2:
+                            if frame != col.name:
+                                frminfo = f' ({col.name})'
+                            lon_col = Column(getattr(col, latlon[3]), name=latlon[1],
+                                             description=f'{frame.capitalize()} Longitude{frminfo}',
+                                             unit=col.representation_component_units[latlon[3]],
+                                             format='.12f')
+                            lat_col = Column(getattr(col, latlon[2]), name=latlon[0],
+                                             description=f'{frame.capitalize()} Latitude{frminfo}',
+                                             unit=col.representation_component_units[latlon[2]],
+                                             format='+.12f')
+                            self.cols.append(lon_col)
+                            self.cols.append(lat_col)
+                            to_pop.append(i)   # Delete original ``SkyCoord`` column.
+
+                # Convert all other ``SkyCoord`` columns that are not in the above three
+                # representations to string valued columns. Those could either be types not
+                # supported yet (e.g. 'helioprojective'), or already present and converted.
+                # If there were any extra ``SkyCoord`` columns of one kind after the first one,
+                # then they have been skipped the decomposition into their component columns.
```

Comment:
I wasn't sure what was meant here, so a minor suggestion! Everything else looks great.

```suggestion
                # then their decomposition into component columns has been skipped.
```